### PR TITLE
Flatten scripts/* and use stderr for errors, add some deprecations

### PR DESCRIPTION
**Pull request by dcousens**
This pull request tidy's up a number of internal modules, flattening their directory structures and removing some now deprecated work-arounds.

As part of the moving, `@keystone-6/core/system` exports (`createSystem`, `createExpressServer`, `initConfig` and `createAdminUIMiddleware`) have been deprecated, and should be removed in the next breaking change.

This pull request additionally changes a number of `console.log`s to `console.error` to unify errors being printed to `stderr`.
Lastly, this pull request fixes `keystone dev` exiting when `db.enableLogging` was an array.

### CODE REVIEW

File: packages/core/src/access.ts [hunk 1]
```diff
@@ -9,7 +10,7 @@ export function denyAll() {
   return false;
 }
 
-export function unfiltered() {
+export function unfiltered<ListTypeInfo extends BaseListTypeInfo>(): MaybePromise<boolean | ListTypeInfo['inputs']['where']> {
```
Comment (dcousens) [inline packages/core/src/access.ts:13]: The type ends up inferring as `=> boolean`, which isn't helpful when you want the type to help infer other function types.

File: packages/core/src/admin-ui/system/buildAdminUI.ts [hunk 2]
```diff
@@ -1,24 +0,0 @@
...
-  const prevNodeEnv = env.NODE_ENV;
-  // Next does a broken build unless we set NODE_ENV to production
-  // @ts-ignore
-  env.NODE_ENV = 'production';
-  try {
-    // importing next/dist/build is quite expensive so we're requiring it lazily
-    /** We do this to stop webpack from bundling next inside of next */
-    const next = 'next/dist/build';
-    const build = require(next).default;
-    await build(projectAdminPath);
```
Comment (dcousens) [inline packages/core/src/admin-ui/system/buildAdminUI.ts:14]: I can't reproduce any behaviors related to this

File: packages/core/src/lib/server/createAdminUIMiddleware.ts [hunk 3]
```diff
@@ -6,21 +6,6 @@ import { pkgDir } from '../../pkg-dir';
...
 const adminErrorHTMLFilepath = path.join(pkgDir, 'static', 'admin-error.html');
 
-type NextApp = {
-  prepare(): Promise<void>;
-  getRequestHandler(): express.Application;
-  render(req: express.Request, res: express.Response, url: string): void;
-};
-
-export async function getNextApp(dev: boolean, projectAdminPath: string): Promise<NextApp> {
-  /** We do this to stop webpack from bundling next inside of next */
```
Comment (dcousens) [inline packages/core/src/lib/server/createAdminUIMiddleware.ts:16]: Cannot reproduce

**Review by borisno2 — APPROVED** [review]
Nice one

**Pull request merged**